homeassistant/components/weather/metoffice.py

```diff
@@ -0,0 +1,117 @@
+"""
+Support for UK Met Office weather service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.metoffice/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.weather import WeatherEntity, PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, TEMP_CELSIUS
+from homeassistant.helpers import config_validation as cv
+# Reuse data and API logic from the sensor implementation
+from homeassistant.components.sensor.metoffice import \
+    MetOfficeCurrentData, CONF_MO_API_KEY, CONF_ATTRIBUTION
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['datapoint==0.4.3']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Required(CONF_MO_API_KEY): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Met Office weather platform."""
+    import datapoint as dp
+    datapoint = dp.connection(api_key=config.get(CONF_MO_API_KEY))
+
+    if None in (hass.config.latitude, hass.config.longitude):
+        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
```

Comment:
Done. User's can specify `latitude` and `longitude` in the component configs, and if they do not it will default to their home `latitude` and `longitude`.

Documentation also updated in home-assistant/home-assistant.github.io#2318.